diff --git a/libopkg/opkg_archive.c b/libopkg/opkg_archive.c
index 1e8c9d1..8c9e8a9 100644
--- a/libopkg/opkg_archive.c
+++ b/libopkg/opkg_archive.c
@@ -423,7 +423,7 @@ static int extract_entry(struct archive *a, struct archive_entry *entry,
         break;
 
     case ARCHIVE_WARN:
-        opkg_msg(NOTICE, "Warning when extracting archive entry '%s': %s\n",
+        opkg_msg(DEBUG, "Warning when extracting archive entry '%s': %s\n",
                  archive_entry_pathname(entry), archive_error_string(a));
         break;
 
