diff -urN a/libopkg/pkg.c b/libopkg/pkg.c
--- a/libopkg/pkg.c	2019-08-11 17:40:36.000000000 +0200
+++ b/libopkg/pkg.c	2019-10-03 11:16:38.842669659 +0200
@@ -752,9 +752,9 @@
             if (pkg->section) {
                 fprintf(fp, "Section: %s\n", pkg->section);
             }
-        } else if (strcasecmp(field, "SHA256sum") == 0) {
+        } else if (strcasecmp(field, "SHA256") == 0) {
             if (pkg->sha256sum) {
-                fprintf(fp, "SHA256sum: %s\n", pkg->sha256sum);
+                fprintf(fp, "SHA256: %s\n", pkg->sha256sum);
             }
         } else if (strcasecmp(field, "Size") == 0) {
             if (pkg->size) {
diff -urN a/libopkg/pkg_parse.c b/libopkg/pkg_parse.c
--- a/libopkg/pkg_parse.c	2019-08-11 17:40:36.000000000 +0200
+++ b/libopkg/pkg_parse.c	2019-10-03 11:16:46.026633360 +0200
@@ -250,8 +250,8 @@
     case 'S':
         if ((mask & PFM_SECTION) && is_field("Section", line))
             pkg->section = parse_simple("Section", line);
-        else if ((mask & PFM_SHA256SUM) && is_field("SHA256sum", line))
-            pkg->sha256sum = parse_simple("SHA256sum", line);
+        else if ((mask & PFM_SHA256SUM) && is_field("SHA256", line))
+            pkg->sha256sum = parse_simple("SHA256", line);
         else if ((mask & PFM_SIZE) && is_field("Size", line)) {
             char *tmp = parse_simple("Size", line);
             pkg->size = strtoul(tmp, NULL, 0);
